### DISCUSSION TITLE
How to connect API endpoints using Basic Authentication?

### DISCUSSION THREAD

**Discussion by mk-khan** [provectus/kafka-ui#2459] [Q&A] [answered]
Hi Team, 

I am in need to connect to api endpoint and get the same response that is shown on the GET consumer group information, I have basic authentication turned on. 

I tried to use post-man to connect to API endpoint, with basic auth username and password

GET: {{baseurl}}/kafkaui/api/clusters/kafka/consumer-groups/connect-topic1

and it is redirecting it to LOGIN FORM page instead of returning the data for the request. I tried to check whats happening and it seems thats how SPRING boot is handing the auth for the kafkaui.

Could you please let me know, if its possible to access API endpoints without going through FORM_LOGIN. 

I came across below think, which seems to handle both API and FORM_LOGIN. 
[https://stackoverflow.com/questions/33739359/combining-basic-authentication-and-form-login-for-the-same-rest-api](https://stackoverflow.com/questions/33739359/combining-basic-authentication-and-form-login-for-the-same-rest-api)

Any help is much appreciated.

**Comment by Haarolean** [comment, marked as answer]
Hi,

upon sending the following request:
`curl -i 'http://localhost:8080/auth' --data-raw 'username=admin&password=admin'`
you get this response with a cookie:
```
HTTP/1.1 302 Found
...
Set-Cookie: SESSION=e740e662-f460-4848-943f-72df4a31e090; Path=/; HttpOnly; SameSite=Lax
content-length: 0
```

Set this cookie for the future requests.

**Reply by mk-khan** [reply]
Thanks @Haarolean , this should be good enough for now. Is there a time limit on this cookie, that i need to handle?

**Reply by Haarolean** [reply]
There's no max age by default.
